crates/sui-package-management/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Context};
+use std::path::PathBuf;
+
+use move_package::lock_file::{self, LockFile};
+use sui_json_rpc_types::{get_new_package_obj_from_response, SuiTransactionBlockResponse};
+use sui_sdk::wallet_context::WalletContext;
+
+pub enum LockCommand {
+    Publish,
+    Upgrade,
+}
+
+/// Update the `Move.lock` file with automated address management info.
+/// Expects a wallet context, the publish or upgrade command, its response.
+/// The `Move.lock` principally file records the published address (i.e., package ID) of
+/// a package under an environment determined by the wallet context config. See the
+/// `ManagedPackage` type in the lock file for a complete spec.
+pub async fn update_lock_file(
+    context: &WalletContext,
+    command: LockCommand,
+    install_dir: Option<PathBuf>,
+    lock_file: Option<PathBuf>,
+    response: &SuiTransactionBlockResponse,
+) -> Result<(), anyhow::Error> {
+    let chain_identifier = context
+        .get_client()
+        .await
+        .context("Network issue: couldn't use client to connect to chain when updating Move.lock")?
+        .read_api()
+        .get_chain_identifier()
+        .await
+        .context("Network issue: couldn't determine chain identifier for updating Move.lock")?;
+
+    let (original_id, version, _) = get_new_package_obj_from_response(response).context("Expected a valid published package response but didn't see one when attempting to update the `Move.lock`.")?;
+    let Some(lock_file) = lock_file else {
+        bail!("Expected a `Move.lock` file to exist after publishing package, but none found. Consider running `sui move build` to generate the `Move.lock` file in the package directory.")
```

Comment:
nit: manually reflow the string to be a bit shorter? You can escape newlines and leading whitespace in Rust strings with a backslash:

```rust
bail!(
    "A really long message \
     can be broken while \
     preserving its content."
);
```